pandas/tests/arrays/datetimes/test_constructors.py

```diff
@@ -232,7 +233,7 @@ def test_from_arrowtest_from_arrow_with_different_units_and_timezones_with_(
     dtype = DatetimeTZDtype(unit=pd_unit, tz=pd_tz)
 
     result = dtype.__from_arrow__(arr)
-    expected = DatetimeArray._from_sequence(
+    expected = DatetimeArray._simple_new(
```

Comment:
I think this was changed from the regular ``DatetimeArray`` constructor to ``DatetimeArray._from_sequence``. 

If this patch is correct, than that would've been wrong and ``_simple_new`` should also be the correct replacement here.

Reply:
Ideally we would use a different constructor, I think (maybe just `pd.array(..., dtype=dtype)`?), because otherwise we are testing that `__from_arrow__` which uses `_simple_new` gives the same result as ``_simple_new`` ... (which was now happening with `_from_sequence`)

Reply:
Good point.  Could do something like `DatetimeArray._from_sequence(int64_data, dtype=f"M8[{unit}]").tz_localize("UTC").astype(self, copy=False)`

Once pyarrow is required we can also make DTA._from_sequence Just Work with the pyarrow object as input